### ISSUE TITLE
Add Kaggle notebook version

### ISSUE THREAD

**Issue by dimka11** [Kaggle/kaggle-api#525]
Please consider to add the feature for specify the notebook version to download the output.

**Comment by osbm** [comment]
I need this so much. This would be an awesome addition.

**Comment by ethanknights** [comment]
Yeah agree that this is still a much needed request, particularly in light of the newer kaggle-benchmarks feature. @Pran-Ker - that seems like a clever workaround but frustrating that we need to mimic a user to access these - in fact for what I'm trying I don't think its possible in browser (or maybe I'm missing it!).

I thought I'd be able to download outputs from unique kaggle-benchmark model runs, but they don't seem to be exposed in the CLI for this same reason, where we can't specify a notebook/run version.

<img width="1505" height="760" alt="Image" src="https://github.com/user-attachments/assets/6def4ed2-1fbc-4656-bf04-862fa33955ef" />
- I can't seem to select any except kernel outputs except from the latest run (qwen at the top).



Edit: Maybe this will draft PR will help in the future :) https://github.com/Kaggle/kaggle-cli/pull/955
